src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp

```diff
@@ -3824,6 +3850,88 @@ void MusicXMLParserDirection::handleRepeats(Measure* measure, const track_idx_t
     }
 }
 
+Marker* MusicXMLParserDirection::findMarker(const String& repeat, bool& coda, SegnoStack& segnos, const Fraction& tick)
+{
+    Marker* m = nullptr;
+    if (repeat == u"segno") {
+        if (Marker* prevSegno = muse::value(segnos, tick.ticks(), nullptr)) {
+            // If there is another segno at this tick, combine their text and increase the label
+            // eg. two segno symbols will have the label 'segnoii'
+            if (prevSegno->xmlText().contains(u"<sym>segno</sym>") && m_wordsText.contains(u"<sym>segno</sym>")) {
+                prevSegno->setXmlText(prevSegno->xmlText() + u" " + m_wordsText);
+                m_wordsText.clear();
+                const int segnoCount = prevSegno->xmlText().count(u"<sym>segno</sym>");
+                const String label = String::fromStdString(std::string(segnoCount, 'i'));
+                prevSegno->setLabel(prevSegno->label() + label);
+            }
+        } else {
+            m = Factory::createMarker(m_score->dummy());
+            // note: Marker::read() also contains code to set text style based on type
+            // avoid duplicated code
+            // apparently this MUST be after setTextStyle
+            m->setMarkerType(MarkerType::SEGNO);
+            m->setLabel(m->label() + m_segnoId);
+            segnos.insert(std::pair<int, Marker*>(tick.ticks(), m));
```

Comment:
It would be great not to change this map here, however, I'm not sure it's easy to do...